SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py

```diff
@@ -416,9 +428,10 @@ def customise_Reco(process,pileup):
     process.pixelTracks.FilterPSet.chi2 = cms.double(50.0)
     process.pixelTracks.FilterPSet.tipMax = cms.double(0.05)
     process.pixelTracks.RegionFactoryPSet.RegionPSet.originRadius =  cms.double(0.02)
-    process.templates.DoLorentz=False
-    process.templates.LoadTemplatesFromDB = cms.bool(False)
-    process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)
+    # use defaults d.k. 2/16
+    #process.templates.DoLorentz=False
+    #process.templates.LoadTemplatesFromDB = cms.bool(False)
+    #process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)

```

Comment:
Matti,
Sure, I can do that too. Last time I looked I did not see any “era” implementations for pixels so
I thought it did not exist.

I am not sure how this works, I assume :
1) everything in block “SiPixelSimBlock” is taken as default ?
2) things which are in  the “def” sections override the default, so I only need to put there things which are
different?
3) What is the difference between Phase1 and Phase1WithPileup?
4) How do I test it? I mean how do I switch between different “def” sections?

For the default part I would suggest to empty the “DeadModules” part.
It is useless.

I would still like to keep the modified customs since some people might be using them.

Best regards,
Danek


On 9 Feb 2016, at 13:02, Matti Kortelainen <notifications@github.com<mailto:notifications@github.com>> wrote:


In SLHCUpgradeSimulations/Configuration/python/phase1TkCustoms.py<https://github.com/cms-sw/cmssw/pull/13228#discussion_r52298966>:

> @@ -416,9 +428,10 @@ def customise_Reco(process,pileup):
>      process.pixelTracks.FilterPSet.chi2 = cms.double(50.0)
>      process.pixelTracks.FilterPSet.tipMax = cms.double(0.05)
>      process.pixelTracks.RegionFactoryPSet.RegionPSet.originRadius =  cms.double(0.02)
> -    process.templates.DoLorentz=False
> -    process.templates.LoadTemplatesFromDB = cms.bool(False)
> -    process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)
> +    # use defaults d.k. 2/16
> +    #process.templates.DoLorentz=False
> +    #process.templates.LoadTemplatesFromDB = cms.bool(False)
> +    #process.PixelCPEGenericESProducer.useLAWidthFromDB = cms.bool(False)


@dkotlins<https://github.com/dkotlins> Could you please apply these changes here https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/SimGeneral/MixingModule/python/SiPixelSimParameters_cfi.py#L37 instead? (I would suggest to use e.g. 800pre6) After #12806<https://github.com/cms-sw/cmssw/pull/12806> that would be the proper place (ok, it doesn't harm to apply the changes also here, but maybe it is not worth of the effort to have the changes in two places).

Said that I'm starting to think if there is really any need to keep the already-migrated-to-era pieces in phase1TkCustoms.py. I guess the question is then if any piece of phase1TkCustoms.py will be needed for phase2 (tracker/tracking), or will phase2 be built from scratch using eras? @boudoul<https://github.com/boudoul> @ebrondol<https://github.com/ebrondol> do you have any idea?

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/13228/files#r52298966>.